.circleci/config.yml

```diff
@@ -0,0 +1,198 @@
+version: 2
+jobs:
+  build:
+    machine:
+      image: circleci/classic:201808-01
+    working_directory: /tmp/src/nitransforms
+    environment:
+      TZ: "/usr/share/zoneinfo/America/Los_Angeles"
+      SCRATCH: "/scratch"
+
+    steps:
+    - restore_cache:
+        keys:
+          - build-v1-{{ .Branch }}-{{ epoch }}
+          - build-v1-{{ .Branch }}-
+          - build-v1-master-
+          - build-v1-
+        paths:
+          - /tmp/docker
+    - run:
+        name: Set-up a Docker registry
+        command: |
+            docker run -d -p 5000:5000 --restart=always --name=registry \
+                -v /tmp/docker:/var/lib/registry registry:2
+    - run:
+        name: Pull existing images
+        command: |
+            set +e
+            docker pull localhost:5000/ubuntu
+            success=$?
+            set -e
+            if [[ "$success" = "0" ]]; then
+                echo "Pulling from local registry"
+                docker tag localhost:5000/ubuntu ubuntu:xenial-20200114
+                docker pull localhost:5000/nitransforms
+                docker tag localhost:5000/nitransforms nitransforms:latest
+            else
+                echo "Pulling from Docker Hub"
+                docker pull ubuntu:xenial-20200114
+                docker tag ubuntu:xenial-20200114 localhost:5000/ubuntu
+                docker push localhost:5000/ubuntu
+            fi
+    - checkout
+    - run:
+        name: Build Docker image & push to registry
+        no_output_timeout: 60m
+        command: |
+            e=1 && for i in {1..5}; do
+                docker build --rm --cache-from=nitransforms:latest \
+                -t nitransforms:latest \
+                --build-arg BUILD_DATE=`date -u +"%Y-%m-%dT%H:%M:%SZ"` \
+                --build-arg VCS_REF=`git rev-parse --short HEAD` . \
+                && e=0 && break || sleep 15
+            done && [ "$e" -eq "0" ]
+            docker tag nitransforms:latest localhost:5000/nitransforms
+            docker push localhost:5000/nitransforms
+    - run:
+        name: Docker registry garbage collection
+        command: |
+            docker exec -it registry /bin/registry garbage-collect --delete-untagged \
+                /etc/docker/registry/config.yml
+    - save_cache:
+        key: build-v1-{{ .Branch }}-{{ epoch }}
+        paths:
+          - /tmp/docker
+    - run:
+        name: Store FreeSurfer license file
+        command: |
+            mkdir -p /tmp/fslicense
+            cd /tmp/fslicense
+            echo "cHJpbnRmICJrcnp5c3p0b2YuZ29yZ29sZXdza2lAZ21haWwuY29tXG41MTcyXG4gKkN2dW12RVYzelRmZ1xuRlM1Si8yYzFhZ2c0RVxuIiA+IGxpY2Vuc2UudHh0Cg==" | base64 -d | sh
+    - persist_to_workspace:
+        root: /tmp
+        paths:
+          - fslicense
+
+  test_pytest:
```

Comment:
why not just pulling fmriprep's image as done in travis?

Reply:
this removes a fair chunk of bloat, plus adds some dependencies not found in fmriprep's image.